clients/issue.go

```diff
@@ -18,6 +18,15 @@ import "time"
 
 // Issue represents a thread like GitHub issue comment thread.
 type Issue struct {
-	URI       *string
-	UpdatedAt *time.Time
+	URI               *string
+	CreatedAt         time.Time
+	ClosedAt          time.Time
+	AuthorAssociation string
+	Comments          []IssueComment
+}
+
+// IssueComment represents a comment on an issue.
+type IssueComment struct {
+	CreatedAt         time.Time
+	AuthorAssociation string
```

Comment:
Nit: maybe update `AuthorAssociation` to be of enum type so that different implementations of the `RepoClient` interface can return valid author associations.

Reply:
Yeah, this makes the code nicer. I'm glad you suggested it.